### PR TITLE
automatic translations added

### PR DESCRIPTION
**Pull request by dokato**
# Automatic translations added

For now with google cloud api, but implemented in a way that it should be possible to use other services, eg. Microsoft, yandex, etc.

### CODE REVIEW

**Comment by filipstachura** [comment]
Very interesting and cool! 🎉 

It would be great to use it during user tests. We start with automatic translation and gradually improve translations file. We fall back to automatic for anything that is not specified in the file.

This process would allow for very fast customer development and keep iterations very fast.

What do you think about it?

**Comment by dokato** [comment]
> What do you think about it?

Sure. That could be easily done with how the things are implemented now. We could introduce some option `automatic.supported` (tranlsation) that used `translate` method first, but when it does not find a translation there it throws a warning and does `automatictranslate` then.

File: R/translator.R [hunk 1]
```diff
@@ -117,5 +127,19 @@ Translator$methods(
       # TODO numbers parsing
       warning("This is not implemented yet. Sorry!")
       number
+    },
+    automatictranslate = function(keyword, api = "google") {
```
Comment (krystian8207) [inline R/translator.R:131]: Let's use consistent names: `automatic_translate`

File: R/translator.R [hunk 2]
```diff
@@ -83,6 +87,8 @@ Translator$methods(
     },
     translate = function(keyword) {
       "Translates 'keyword' to language specified by 'set_translation_language'"
+      if (isTRUE(automatic))
+        stop("For automatic translations use 'automatictranslate' or 'at'.")
```
Comment (krystian8207) [inline R/translator.R:91]: Does it mean we cannot use both custom translations and google ones? I think it's an unnecessary restriction. WDYT?

File: R/translator.R [hunk 3]
```diff
@@ -117,5 +127,19 @@ Translator$methods(
...
+      using cloud service 'api'"
+      if (identical(translation_language, character(0)))
+        stop("Please provide a 'translation_language'. Check docs how.")
+      tr <- switch(api,
+       google = translate_with_google_cloud(keyword, translation_language),
+       stop("This 'api' is not supported.")
+      )
+    },
+    at = function(keyword) {
+      "Wrapper method. Look at 'translate'"
```
Comment (krystian8207) [inline R/translator.R:142]: -> "Wrapper method. Look at 'automatictranslate' ?

File: DESCRIPTION [hunk 4]
```diff
@@ -23,4 +23,5 @@ Suggests:
     knitr,
     rmarkdown,
     testthat,
-    covr
+    covr,
+    googleLanguageR
```
[the remark this answers is not in the document]
Comment (dokato) [reply DESCRIPTION:27]: It is in the Suggestion, isn't it?

File: R/translator.R (hunk 2 above)
Comment (dokato) [reply R/translator.R:91]: agreed

**Pull request merged**